### DISCUSSION TITLE
Feature Request: Enhanced Electricity Spot Price Monitoring

### DISCUSSION THREAD

**Discussion by heinemannj** [JensenNick/huawei_solar_pees#64] [Forum]
I'm with Tibber and living in Germany:

![image](https://github.com/JensenNick/huawei_solar_pees/assets/3251642/cac18de8-5c40-402a-8a66-9102637d58a0)

![image](https://github.com/JensenNick/huawei_solar_pees/assets/3251642/b4caafb1-d30c-4da0-8a61-0c564e8fc7ce)

![image](https://github.com/JensenNick/huawei_solar_pees/assets/3251642/57bdbabd-06f3-4581-a29d-2da764c0844d)

![image](https://github.com/JensenNick/huawei_solar_pees/assets/3251642/9c355c78-f534-43ed-92d0-c5b6f1489d92)

![image](https://github.com/JensenNick/huawei_solar_pees/assets/3251642/35b97193-f234-4850-bbb9-1b36c86efe6f)

**My approach**: **To be much close to the nameing conventions provided by @wlcrs huawei_solar** and all other integrations I needs to synchronize:

- huawei_solar
- epexspot
- tibber
- sun
- sun2
- solcast
- emhass
- myVaillant
- dwd_DeutscherWetterdienst
- Shelly
- ...

It works as **visible in above screenshoots**!

For sure tons of bugs and a lot of improvement - but I need clear readable and short sensor names for proper debugging on all integration dependencies ...

**Comment by heinemannj** [comment]
I've made a quick and dirty review of your code.

My view:

- Very complex
- Hardly to maintain
- Not closely enough to @wlcrs naming convention

My conclusion:

- I will step out

**Sorry for the noise ...**

**Comment by JensenNick** [comment]
What is the issue implementing the pricing from Tibber? All you need for PEES to work properly is an sensor providing you with the current price and one with the current sales price. It would be way to ambitious to try to implement and maintain all the various electricity providers pricing models etc. This is IMO better kept outside the PEES package.
